packages/modules/viewer-react/src/components/app-ui/EcsqlWidget.tsx

```diff
@@ -0,0 +1,141 @@
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { UiFramework } from "@itwin/appui-react";
+import { QueryRowFormat } from "@itwin/core-common";
+import {
+  Button,
+  ExpandableBlock,
+  LabeledTextarea,
+  Table,
+} from "@itwin/itwinui-react";
+import React, { useState } from "react";
+
+export default function EcsqlWidget() {
+  const [input, setInput] = useState<string>("");
+  const [rows, setRows] = useState<Record<string, string>[]>([]);
+  const [headers, setHeaders] = useState<string[]>([]);
+  const [isloading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string>("");
+
+  const getData = async () => {
+    try {
+      setRows([]);
+      setHeaders([]);
+      setLoading(true);
+      const results = UiFramework.getIModelConnection()!.query(
+        input,
+        undefined,
+        { rowFormat: QueryRowFormat.UseJsPropertyNames }
+      );
+      const rows = [];
+      const headers: string[] = [];
+
+      for await (const obj of results) {
+        const row = { ...obj };
+        // Case 1: Need to convert booleans to strings to display them in the table
+        Object.keys(row)
+          .filter((key) => typeof row[key] === "boolean")
+          .forEach((key) => {
+            row[key] = obj[key].toString();
+          });
+        // Case 2: Need to stringify objects to display them in the table
+        Object.keys(row)
+          .filter((key) => typeof row[key] === "object")
+          .forEach((key) => {
+            row[key] = JSON.stringify(obj[key]);
+          });
+        // Case 3: If the cell content is too long, put it inside an ExpandableBlock
+        Object.keys(row)
+          .filter(
+            (key) => typeof row[key] === "string" && row[key].length > 500
+          )
+          .forEach((key) => {
+            row[key] = (
+              <ExpandableBlock title="..." size="small" styleType="borderless">
+                {row[key]}
+              </ExpandableBlock>
+            );
+          });
+
+        rows.push(row);
+        for (const key of Object.keys(row)) {
+          if (!headers.includes(key)) {
+            headers.push(key);
+          }
+        }
+      }
+      setRows(rows);
+      setHeaders(headers);
+      setLoading(false);
+      setError("");
+    } catch (err: any) {
+      setLoading(false);
+      setError(err.message);
+    }
+  };
+
+  const columns = React.useMemo(
+    () => [
+      {
+        Header: "Header name",
+        columns: headers.map((str) => ({
+          id: str,
+          Header: str,
+          accessor: str,
+          width: 300,
+        })),
+      },
+    ],
+    [headers]
+  );
+  const data = rows;
+
+  return (
+    <div style={{ height: "100%", display: "flex", flexFlow: "column" }}>
```

Comment:
for now inline styles are fine, and you can leave this as is, but just a heads up we prefer using scss files